kubermatic-addons/custom-addon/flux2/01-bootstrap.yaml

```diff
@@ -0,0 +1,45 @@
+{{- $httpCredentialsProvided := and (not (empty .Variables.gitRepositoryHttpUsername)) (not (empty .Variables.gitRepositoryHttpPassword)) -}}
+{{- $authenticationRequired := $httpCredentialsProvided -}}
+---
+{{- if $authenticationRequired }}
```

Comment:
I don't see obvious reason why you have two variables `httpCredentialsProvided` and `authenticationRequired `

I guess the situation is that either both are provided and secret is created, otherwise it's not ... correct?

Reply:
@michalvanco indeed, this construct is currently not necessary. The idea was to provide an easy way to extend the addon in the future e.g. with SSH based authentication. Only if we have more than one authentication method this would make sense, so we can create the correct fields in the secret (username/password vs. identity/indentity.pub/...). Should we keep it or drop it for simplicity?

Reply:
unless you want to provide the ssh support right away (I would it's not needed as most of the providers will work nicely with the TOKEN auth), I would just drop these extra variables and make that simple

as you say, later on we can always extend it with such logic